### DISCUSSION TITLE
[Hackathon Use Case] - FDC3 Share Identity between applications

### DISCUSSION THREAD

**Discussion by Yannick-Malins** [finos/finos-pmcs#258] [Ideas]
# FDC3 Share Identity between applications

## Business Problem

Implementing SSO for every application can be a pain, especially if these applications are running in various different containers (standalone, browser, fdc3 container...)

We already have highly secure applications, such as Symphony, running on the user's desktop. Why don't we leverage FDC3 to allow one FDC3 application to share the user's identity with others?

## Expected Features

- Mock up an FDC3-enabled webapp+backend that requires users to log in
- Add an option to "pull identity from FDC3", which will trigger a new "GetIdentity" intent
- Extend Symphony to process this intent, trigger a dialog allowing the user to share their identity over FDC3
- Implement a basic token+certificate based verification system between the calling app and Symphony (aka "oauth-ish")

## Useful apps, systems, projects, technologies, libraries and datasets

- Symphony
- An FDC3 desktop agent or container

**Comment by robmoffat** [comment]
Some context: https://github.com/finos/FDC3/issues/684

**Comment by Yannick-Malins** [comment]
participants:
https://github.com/Yannick-Symphony
https://github.com/ystxn
https://github.com/tzhao-symphony
https://github.com/antoinerollindev
https://github.com/yinan-symphony

**Comment by Yannick-Malins** [comment]
https://drive.google.com/file/d/1N05MeL6IkKhOsG8SqmdVMRJEVUcJL3LK/view?usp=sharing

**Comment by TheJuanAndOnly99** [comment]
Closing discussion in preparation for [this years hackathon](https://www.finos.org/hosted-events/2024-10-25-citi-india-hackathon-supported-by-finos). Feel free to reopen and update if its still relevant.
